### PR TITLE
feat: [CDS-78064]: Improve coverage pipeline-service : NodeExecutionServiceImpl.java

### PR DESCRIPTION
**Pull request by ayushitiwari024**
## Describe your changes

Added  testcases for the functions that were not covered in NodeExecutipnServiceImpl. Made a private function default for added a annotation visibilityForTesting

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/52398)

### CODE REVIEW

**Comment by ayushitiwari024** [comment]
trigger uts

File: pipeline-service/modules/orchestration/src/main/java/io/harness/engine/executions/node/NodeExecutionServiceImpl.java [hunk 1]
```diff
@@ -296,7 +296,8 @@ public CloseableIterator<NodeExecution> fetchChildrenNodeExecutionsIterator(
     return nodeExecutionReadHelper.fetchNodeExecutions(query);
   }
 
-  private CloseableIterator<NodeExecution> fetchChildrenNodeExecutionsIteratorWithoutProjection(
+  @VisibleForTesting
```
Comment (ankittiwari-harness) [inline pipeline-service/modules/orchestration/src/main/java/io/harness/engine/executions/node/NodeExecutionServiceImpl.java:299]: why add visibleForTesting for a public method

Comment (ayushitiwari024) [reply pipeline-service/modules/orchestration/src/main/java/io/harness/engine/executions/node/NodeExecutionServiceImpl.java:299]: It was a private function for which I had to write a UT case therefor made it public and added the annotation.

**Comment by ayushitiwari024** [comment]
trigger messagecheck

File: pipeline-service/modules/orchestration/src/main/java/io/harness/engine/executions/node/NodeExecutionServiceImpl.java (hunk 1 above)
Comment (ayushitiwari024) [reply pipeline-service/modules/orchestration/src/main/java/io/harness/engine/executions/node/NodeExecutionServiceImpl.java:299]: removed private keyword

**Comment by ayushitiwari024** [comment]
trigger uts

File: pipeline-service/modules/orchestration/src/test/java/io/harness/engine/executions/node/NodeExecutionServiceImplTest.java [hunk 2]
```diff
@@ -881,6 +911,232 @@ public void testFetchNodeExecutionsForGivenStageFQNs() {
...
+    Query query = query(where(NodeExecutionKeys.uuid).is(nodeExecutionId));
+    doReturn(null).when(mongoTemplateMock).findAndModify(query, ops, NodeExecution.class);
+    boolean result = nodeExecutionService.markRetried(nodeExecutionId);
+    assertThat(result).isFalse();
+  }
+
+  @Test
+  @Owner(developers = AYUSHI_TIWARI)
+  @Category(UnitTests.class)
+  public void testMarkRetriedWhenNodeExecutionNotNull() { // different file
```
Comment (prashantsharma04) [inline pipeline-service/modules/orchestration/src/test/java/io/harness/engine/executions/node/NodeExecutionServiceImplTest.java:1098]: please remove this comment // different file

Comment (ayushitiwari024) [reply pipeline-service/modules/orchestration/src/test/java/io/harness/engine/executions/node/NodeExecutionServiceImplTest.java]: addressed

**Review by prashantsharma04 — DISMISSED** [review]

**Review by prashantsharma04 — COMMENTED** [review]
> Reviewed 1 of 1 files at r4, 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52398)* status: 2 file(s) to review. 2. unresolved discussions. 1 of 2 approvals obtained.

**Comment by prashantsharma04** [comment]
trigger uts

**Comment by ayushitiwari024** [comment]
trigger ut6

**Comment by ayushitiwari024** [comment]
trigger uts

File: pipeline-service/modules/orchestration/src/main/java/io/harness/engine/executions/node/NodeExecutionServiceImpl.java [hunk 3]
```diff
@@ -211,6 +211,7 @@ public List<NodeExecution> getAllWithFieldIncluded(Set<String> nodeExecutionIds,
     }
     return mongoTemplate.find(query, NodeExecution.class);
   }
+
```
Comment (prashantsharma04) [inline pipeline-service/modules/orchestration/src/main/java/io/harness/engine/executions/node/NodeExecutionServiceImpl.java:214]: nit: this can be avoided

**Review by prashantsharma04 — APPROVED** [review]

**Review by prashantsharma04 — COMMENTED** [review]
> Reviewed 2 of 3 files at r6, 1 of 1 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52398)* status: 3 file(s) to review. 2. unresolved discussions. 1 of 2 approvals obtained.

**Comment by ayushitiwari024** [comment]
trigger review

**Review by sandeshsalunkhegh — APPROVED** [review]
> Reviewed 2 of 3 files at r6, 1 of 1 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52398)* status: 2. unresolved discussions.

**Pull request merged**